From 00a456145531d194d3993c9f4cd404d5ca16c9df Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 6 Apr 2015 17:36:44 +0000
Subject: [PATCH] quota: Fix build with musl

Include fcntl.h to pacify compiler errors on musl
like

error: unknown type name 'loff_t'
Cover rpc headers under proper conditional
Dont use __P its undefined

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 quotacheck.c    | 1 +
 quotaio.c       | 1 +
 rquota_client.c | 4 ++++
 3 files changed, 6 insertions(+)

diff --git a/quotacheck.c b/quotacheck.c
index bd62d9a..772a27d 100644
--- a/quotacheck.c
+++ b/quotacheck.c
@@ -19,6 +19,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <errno.h>
+#include <fcntl.h>
 
 #include <sys/stat.h>
 #include <sys/types.h>
diff --git a/quotaio.c b/quotaio.c
index 94ae458..d57fc1a 100644
--- a/quotaio.c
+++ b/quotaio.c
@@ -12,6 +12,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <stdlib.h>
+#include <fcntl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/file.h>
diff --git a/rquota_client.c b/rquota_client.c
index 7f8e821..d48505a 100644
--- a/rquota_client.c
+++ b/rquota_client.c
@@ -19,7 +19,9 @@
 
 #include "config.h"
 
+#if defined(RPC)
 #include <rpc/rpc.h>
+#endif
 #include <sys/types.h>
 #include <sys/param.h>
 #include <sys/stat.h>
@@ -35,7 +37,9 @@
 #include <stdint.h>
 
 #include "mntopt.h"
+#if defined(RPC)
 #include "rquota.h"
+#endif
 #include "common.h"
 #include "quotaio.h"
 #include "quotasys.h"
